src/settings.mli

```diff
@@ -14,3 +14,9 @@ val create :
   -> 'a t
 
 val string : scope:ConfigurationTarget.t -> key:string -> string t
+
+(** replace ${workspaceFolder} with the first workspace folder that is open *)
```

Comment:
That does not match the semantics described [here](https://code.visualstudio.com/docs/editor/multi-root-workspaces):

> used in a configuration (for example ${workspaceFolder} or the now deprecated ${workspaceRoot}) are resolved relative to the folder they belong to. It is possible to scope a variable per workspace folder by appending the root folder's name to a variable (separated by a colon).

Reply:
This is really annoying since the VSCode API doesn't provide a way to get the workspace folder of the configuration file.

Since the sandbox configurations are primarily changed through commands, we could do what the C++ and CMake extension do and only support named workspace folder variables (`${workspaceFolder:name}`):

https://github.com/microsoft/vscode-cpptools/blob/839eefa0325a718b7feefc737eaf9d3bdae18a6a/Extension/src/common.ts#L339

https://github.com/microsoft/vscode-cmake-tools/blob/6b30c8f0812c5a5fcc2458544991a200c05d9491/src/expand.ts#L112

Eventually I would hope that VSCode gets native support for settings variables: https://github.com/microsoft/vscode/issues/2809.

Reply:
Indeed, the api doesn't seem well thought out.

> Since the sandbox configurations are primarily changed through commands, we could do what the C++ and CMake extension do and only support named workspace folder variables (${workspaceFolder:name}):

This workaround sounds good to me.